server/src/projects/project-applicants/project-applicant.controller.ts

```diff
@@ -0,0 +1,85 @@
+import { Controller, Patch, Body, Param, Post, UseInterceptors, UseGuards, UsePipes, Delete, HttpException, HttpStatus } from '@nestjs/common';
+import { pick } from 'underscore';
+import { CrmService } from '../../crm/crm.service';
+import { JsonApiSerializeInterceptor } from '../../json-api-serialize.interceptor';
+import { AuthenticateGuard } from '../../authenticate.guard';
+import { JsonApiDeserializePipe } from '../../json-api-deserialize.pipe';
+import { PROJECTAPPLICANT_ATTRS } from './project-applicants.attrs';
+import { CONTACT_ATTRS } from '../../contact/contacts.attrs';
+
+const ACTIVE_CODE = 1;
+
+@UseInterceptors(new JsonApiSerializeInterceptor('project-applicants', {
+  id: 'dcp_projectapplicantid',
+  attributes: [
+    ...PROJECTAPPLICANT_ATTRS,
+
+    'contact',
+  ],
+
+  contact: {
+    ref: 'contactid',
+    attributes: CONTACT_ATTRS,
+  },
+}))
+@UseGuards(AuthenticateGuard)
+@UsePipes(JsonApiDeserializePipe)
+@Controller('project-applicants')
+export class ProjectApplicantController {
+  constructor(private readonly crmService: CrmService) {}
+
+  @Post('/')
+  async create(@Body() body) {
+    const allowedAttrs = pick(body, PROJECTAPPLICANT_ATTRS);
+    const email = body.emailaddress;
+    const projectId = body.project;
+    const fullName = body.dcp_name;
+
+    let contactId;
+
+    // check if contact already exists for this email address
+    const { records } = await this.crmService.get('contacts', `
+      $select=${CONTACT_ATTRS.join(',')}
+      &$filter=startswith(emailaddress1, '${email}')
+        and statuscode eq ${ACTIVE_CODE}
```

Comment:
Should we find ALL contacts first, and then reactivate a contact if it already exists? 
Here, the code might create a contact that already exists but is just deactivated 

No problem if you wanna do it in a separate PR, though

Reply:
Ooh are `contacts` also able to be activated/deactivated? I thought it was only projectApplicants